BigQuery Java API: NPE on com.google.cloud.bigquery.StandardTableDefinition.fromPb(StandardTableDefinition.java:298)

#### Environment details

1. Specify the API at the beginning of the title (for example, "BigQuery: ...") : BigQuery
   General, Core, and Other are also allowed as types
2. OS type and version:  Any OS
3. Java version: 1.8
4. google-cloud-java version(s): 
            <groupId>com.google.cloud</groupId>
            <artifactId>google-cloud-bigquery</artifactId>
            <version>1.110.0</version>

#### Steps to reproduce

**NOT ABLE TO REPRODUCE**

#### Code example

```java

Page<Table> tables = bigquery.listTables(datasetId, BigQuery.TableListOption.pageSize(1));
Iterable<Table> table_iterator = tables.iterateAll();
for (Table table : table_iterator){
if (table.getDefinition().getType().toString().equalsIgnoreCase("TABLE")) {
                                           System.out.println("table");
                                        }
}

```

#### Stack trace
```
Caused by: java.lang.NullPointerException: Null pointer - Got unexpected time partitioning {"field":"XXXXX"} in project XXXXXX in dataset XXXXX in table XXXXX java.lang.NullPointerException: Name is null
        at com.google.cloud.bigquery.StandardTableDefinition.fromPb(StandardTableDefinition.java:298)
        at com.google.cloud.bigquery.TableDefinition.fromPb(TableDefinition.java:151)
        at com.google.cloud.bigquery.TableInfo$BuilderImpl.<init>(TableInfo.java:188)
        at com.google.cloud.bigquery.Table.fromPb(Table.java:624)
        at com.google.cloud.bigquery.BigQueryImpl$21.apply(BigQueryImpl.java:847)
        at com.google.cloud.bigquery.BigQueryImpl$21.apply(BigQueryImpl.java:844)
        at com.google.common.collect.Iterators$6.transform(Iterators.java:786)
        at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47)
        at com.google.cloud.PageImpl$PageIterator.computeNext(PageImpl.java:72)
        at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:141)
        at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:136)
```



#### Any additional information below	

This issue occurs on a few tables in Prod which I am not able to replicate in Dev. 
The rpc call has the partition type as null but the TimePartitioning object in the response pb is not null which is leading to an NPE at this point.
https://github.com/googleapis/java-bigquery/blob/b539f95965ddba6ce4fc5b6712ce1e38b8286bd7/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java#L276

--- a/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java
+++ b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java
@@ -285,17 +285,6 @@ public abstract class StandardTableDefinition extends TableDefinition {
                 + " in table "
                 + tablePb.getTableReference().getTableId(),
             e);
-      } catch (NullPointerException e) {
-        throw new NullPointerException(
-            "Null pointer - Got unexpected time partitioning "
-                + tablePb.getTimePartitioning().toString()
-                + " in project "
-                + tablePb.getTableReference().getProjectId()
-                + " in dataset "
-                + tablePb.getTableReference().getDatasetId()
-                + " in table "
-                + tablePb.getTableReference().getTableId()
-                + e.toString());
       }
     }
     if (tablePb.getRangePartitioning() != null) {

--- a/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TimePartitioning.java
+++ b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TimePartitioning.java
@@ -142,7 +142,7 @@ public abstract class TimePartitioning implements Serializable {
     if (Data.isNull(expirationMs)) {
       expirationMs = null;
     }
-    return newBuilder(Type.valueOf(partitioningPb.getType()))
+    return newBuilder(Type.valueOf(firstNonNull(partitioningPb.getType(), Type.DAY.name())))
         .setExpirationMs(expirationMs)
         .setField(partitioningPb.getField())
         .setRequirePartitionFilter(partitioningPb.getRequirePartitionFilter())

--- a/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/BigQueryImplTest.java
+++ b/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/BigQueryImplTest.java
@@ -116,10 +116,21 @@ public class BigQueryImplTest {
   private static final Long TABLE_CREATION_TIME = 1546275600000L;
   private static final TimePartitioning TIME_PARTITIONING =
       TimePartitioning.of(TimePartitioning.Type.DAY, EXPIRATION_MS);
+  private static final com.google.api.services.bigquery.model.TimePartitioning PB_TIMEPARTITIONING =
+      new com.google.api.services.bigquery.model.TimePartitioning()
+          .setType(null)
+          .setField("timestampField");
+  private static final TimePartitioning TIME_PARTITIONING_NULL_TYPE =
+      TimePartitioning.fromPb(PB_TIMEPARTITIONING);
   private static final StandardTableDefinition TABLE_DEFINITION_WITH_PARTITIONING =
       StandardTableDefinition.newBuilder()
           .setSchema(TABLE_SCHEMA)
           .setTimePartitioning(TIME_PARTITIONING)
+          .build();
+  private static final StandardTableDefinition TABLE_DEFINITION_WITH_PARTITIONING_NULL_TYPE =
+      StandardTableDefinition.newBuilder()
+          .setSchema(TABLE_SCHEMA)
+          .setTimePartitioning(TIME_PARTITIONING_NULL_TYPE)
           .build();
   private static final RangePartitioning.Range RANGE =
       RangePartitioning.Range.newBuilder().setStart(1L).setInterval(2L).setEnd(10L).build();
@@ -142,7 +153,10 @@ public class BigQueryImplTest {
       TableInfo.newBuilder(TABLE_ID, TABLE_DEFINITION_WITH_PARTITIONING)
           .setCreationTime(TABLE_CREATION_TIME)
           .build();
-
+  private static final TableInfo TABLE_INFO_WITH_PARTITIONS_NULL_TYPE =
+      TableInfo.newBuilder(TABLE_ID, TABLE_DEFINITION_WITH_PARTITIONING_NULL_TYPE)
+          .setCreationTime(TABLE_CREATION_TIME)
+          .build();
   private static final ModelId OTHER_MODEL_ID = ModelId.of(DATASET, OTHER_MODEL);
   private static final ModelId MODEL_ID_WITH_PROJECT = ModelId.of(PROJECT, DATASET, MODEL);
 
@@ -902,6 +916,22 @@ public class BigQueryImplTest {
     ImmutableList<Table> tableList =
         ImmutableList.of(
             new Table(bigquery, new TableInfo.BuilderImpl(TABLE_INFO_WITH_PARTITIONS)));
+    Tuple<String, Iterable<com.google.api.services.bigquery.model.Table>> result =
+        Tuple.of(CURSOR, Iterables.transform(tableList, TableInfo.TO_PB_FUNCTION));
+    EasyMock.expect(bigqueryRpcMock.listTables(PROJECT, DATASET, TABLE_LIST_OPTIONS))
+        .andReturn(result);
+    EasyMock.replay(bigqueryRpcMock);
+    Page<Table> page = bigquery.listTables(DATASET, TABLE_LIST_PAGE_SIZE, TABLE_LIST_PAGE_TOKEN);
+    assertEquals(CURSOR, page.getNextPageToken());
+    assertArrayEquals(tableList.toArray(), Iterables.toArray(page.getValues(), Table.class));
+  }
+
+  @Test
+  public void testListTablesReturnedParametersNullType() {
+    bigquery = options.getService();
+    ImmutableList<Table> tableList =
+        ImmutableList.of(
+            new Table(bigquery, new TableInfo.BuilderImpl(TABLE_INFO_WITH_PARTITIONS_NULL_TYPE)));
     Tuple<String, Iterable<com.google.api.services.bigquery.model.Table>> result =
         Tuple.of(CURSOR, Iterables.transform(tableList, TableInfo.TO_PB_FUNCTION));
     EasyMock.expect(bigqueryRpcMock.listTables(PROJECT, DATASET, TABLE_LIST_OPTIONS))

--- a/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/StandardTableDefinitionTest.java
+++ b/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/StandardTableDefinitionTest.java
@@ -153,34 +153,6 @@ public class StandardTableDefinitionTest {
   }
 
   @Test
-  public void testFromPbWithNullTimePartitioningTypeRaisesNullPointerException() {
-    Table invalidTable =
-        new Table()
-            .setType("TABLE")
-            .setTableReference(
-                new TableReference()
-                    .setProjectId("NULL_PTR_TEST_PROJECT")
-                    .setDatasetId("NULL_PTR_TEST_DATASET")
-                    .setTableId("NULL_PTR_TEST_TABLE"))
-            .setTimePartitioning(
-                new com.google.api.services.bigquery.model.TimePartitioning().setType(null));
-    try {
-      StandardTableDefinition.fromPb(invalidTable);
-    } catch (NullPointerException ne) {
-      assertThat(
-          ne.getMessage(),
-          allOf(
-              containsString("Null pointer - Got unexpected time partitioning"),
-              containsString("null"),
-              containsString("NULL_PTR_TEST_PROJECT"),
-              containsString("NULL_PTR_TEST_DATASET"),
-              containsString("NULL_PTR_TEST_TABLE")));
-      return;
-    }
-    fail("testFromPb null pointer exception did not throw!");
-  }
-
-  @Test
   public void testFromPbWithNullEstimatedRowsAndBytes() {
     StandardTableDefinition.fromPb(
         TABLE_DEFINITION.toPb().setStreamingBuffer(new Streamingbuffer()));

